File: 4312-limit-occurrences-in-sorted-array/limit-occurrences-in-sorted-array.cs
public class Solution {
    public int[] LimitOccurrences(int[] nums, int k) {
        int count = 0;

        foreach (int num in nums) {
            if (count < k || num != nums[count - k]) {
                nums[count] = num;
                count++;
            }
        }

        int[] ans = new int[count];

        for (int i = 0; i < count; i++) {
            ans[i] = nums[i];
        }

        return ans;
    }
}

File: 4312-limit-occurrences-in-sorted-array/limit-occurrences-in-sorted-array.py
class Solution:
    def limitOccurrences(self, nums: list[int], k: int) -> list[int]:
        counter = Counter(nums)
        ans = []

        for kk, v in counter.items():
            nk = k
            if v <= nk:
                while v:
                    ans.append(kk)
                    v -= 1
            else:
                while nk:
                    ans.append(kk)
                    nk -= 1

        return ans
